pkg/utils/transform/transform.go

```diff
@@ -35,3 +35,15 @@ func InjectDefaultSA(defaultSA string) mf.Transformer {
 		return nil
 	}
 }
+
+func InjectNamespaceCustom(preserveNamespace string, tf mf.Transformer) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
```

Comment:
as we know the `transformer` passed is always `manifetival.InjectNamespace`, we can eliminate the second argument.

before the return statement, let's add 
```
func InjectNamespaceCustom(preserveNamespace string) mf.Transformer {
    tf := mf.InjectNamespace(preserveNamespace)
    return func(u *unstructured.Unstructured) error {
    ...
    ...
```
then use tf in the returned function as it is.

Reply:
Here TargetNamespace is required which comes from `cfg.Spec.TargetNamespace`

Reply:
i see, then can we pass the `targetNamespace` as the second argument to `InjectNamespaceCustom`.

and construct the namespace transformer inside InjectNamespaceCustom.
something like
```
func InjectNamespaceCustom(preserveNamespace, targetNamespace string) mf.Transformer {
    tf := mf.InjectNamespace(targetNamespace)
    return func(u *unstructured.Unstructured) error {
    ...
    ...
```